### PR TITLE
[BugFix] Make sync configurable when writing a new tablet

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Make sync configurable when writing a new tablet.

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/fs/fs_posix.cpp [hunk 1]
```diff
@@ -261,7 +261,9 @@ class PosixWritableFile : public WritableFile {
     Status sync() override {
         if (_pending_sync) {
             _pending_sync = false;
-            RETURN_IF_ERROR(do_sync(_fd, _filename));
+            if (LIKELY(config::sync_tablet_data)) {
```
Comment (chaoyli) [inline be/src/fs/fs_posix.cpp:264]: Why to make it a configuration? Would sync decrease the performance lot?

**Comment by zaorangyang** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by chaoyli — DISMISSED** [review]

File: be/src/fs/fs_posix.cpp [hunk 2]
```diff
@@ -261,7 +261,9 @@ class PosixWritableFile : public WritableFile {
     Status sync() override {
         if (_pending_sync) {
             _pending_sync = false;
-            RETURN_IF_ERROR(do_sync(_fd, _filename));
+            if (config::sync_tablet_meta) {
+                RETURN_IF_ERROR(do_sync(_fd, _filename));
+            }
```
Comment (decster) [inline be/src/fs/fs_posix.cpp:266]: this config should not affect the behavior of posix_fs(it affects all file writes), it should be checked when close segment file I think

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
BE Build Success

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by decster** [comment]
run starrocks_admit_test

**Pull request merged**